cpp/src/fil/infer.cu

```diff
@@ -14,14 +14,26 @@
  * limitations under the License.
  */
 
-#include <algorithm>
-#include <cmath>
+#include "common.cuh"
+
+#include <fil/internal.cuh>
+
+#include <cuml/fil/multi_sum.cuh>
 
 #include <raft/cudart_utils.h>
+
 #include <thrust/functional.h>
-#include <cuml/fil/multi_sum.cuh>
-#include <fil/internal.cuh>
-#include "common.cuh"
+
+#include <algorithm>
+#include <cmath>
+
+#ifndef SAFE_PRAGMA_UNROLL
+#ifdef __CUDA_ARCH__
+#define SAFE_PRAGMA_UNROLL _Pragma("unroll")
```

Comment:
Out of curiosity, why is it "safe"?

Reply:
Perhaps poorly name. I guess because it doesn't result in warnings when you use it in host code (where it is a no-op).

Reply:
I will rename to `CUDA_PRAGMA_UNROLL`

Reply:
Done.